src/bot/mtproto-api/bot/session.ts

```diff
@@ -0,0 +1,47 @@
+import { Context } from "#root/types/context.js";
+import { SupabaseClient } from "@supabase/supabase-js";
+import { StringSession } from "telegram/sessions";
+
+/**
+ * This class extends the StringSession class from the Telegram library.
+ *
+ * It adds the ability to save and load the session data from Supabase.
```

Comment:
This fixes auth? Maybe KV is significantly more lightweight. We might even be able to store in secrets automatically? Spinning up a whole database just for a single string is overkill.

Reply:
Will resolved DB related comments and open an issue for refactoring once merged